x-pack/plugins/fleet/server/routes/agent_policy/index.ts

```diff
@@ -38,135 +40,178 @@ import {
 
 export const registerRoutes = (router: FleetAuthzRouter) => {
   // List - Fleet Server needs access to run setup
-  router.get(
-    {
+  router.versioned
+    .get({
       path: AGENT_POLICY_API_ROUTES.LIST_PATTERN,
-      validate: GetAgentPoliciesRequestSchema,
-      options: { access: 'public' },
       fleetAuthz: {
         fleet: { readAgentPolicies: true },
       },
-    },
-    getAgentPoliciesHandler
-  );
+    })
+    .addVersion(
+      {
+        version: OLDEST_PUBLIC_VERSION,
```

Comment:
Should we rather create a constant e.g. `VERSION_2023_10_31` to use as `addVersion` parameter? We don't want this constant to ever point to another version.